crypto/src/random/mod.rs

```diff
@@ -55,7 +47,8 @@ pub trait RandomCoin: Sync {
     /// PRNG.
     fn draw<E: FieldElement<BaseField = Self::BaseField>>(&mut self) -> Result<E, RandomCoinError>;
 
-    /// Returns a vector of unique integers selected from the range [0, domain_size).
+    /// Returns a vector of unique integers selected from the range [0, domain_size) after it reseeds
+    /// the coin with a nonce by setting the new seed to hash(`seed` || `nonce`).
```

Comment:
nit: I wonder if we should not mention how exactly the reseeding happens as it concavely be done differently for different implementations of this trait.